code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/Token.java

```diff
@@ -0,0 +1,174 @@
+    private final String hostname;
+    private String code;
+    private final String user;
+    private final List<String> orgs;
+    private final List<TeamAccountInfo> teams;
+    private final Boolean security;
+    private final String clientId;
+    private final String userType;
+    private final String authProvider;
+
+    private final String accountId;
+    private final Identity userIdentity;
+    private final boolean enabled;
+    private final List<Identity> identities;
+
+    //LEGACY: Used for old Implementation of projects/ Identities. Remove when vincent changes to new api.
+    public Token(String jwt, String username, List<String> orgs, List<TeamAccountInfo> teams, Boolean security, String clientId, String userType,
+                 String authProvider, String accountId, List<Identity> identities) {
+        this.jwt = jwt;
+        this.user = username;
+        this.authProvider = authProvider;
+        this.hostname = null;
+        this.orgs = orgs;
+        this.teams = teams;
+        this.security = security;
+        this.clientId = clientId;
+        this.userType = userType;
+        this.accountId = accountId;
+        this.userIdentity = null;
+        this.enabled = this.security;
+        this.identities = identities;
+
+    }
+
+    //LEGACY: Used for old Implementation of projects/ Identities. Remove when vincent changes to new api.
+    public Token(Boolean security, String clientId, String hostName, String authProvider) {
+        this.authProvider = authProvider;
+        this.jwt = null;
+        this.user = null;
+        this.orgs = null;
+        this.teams = null;
+        this.security = security;
+        this.clientId = clientId;
+        this.userType = null;
+        this.accountId = null;
+        this.hostname = hostName;
+        this.userIdentity = null;
+        enabled = this.security;
+        identities = null;
+    }
+
+    public Token(String jwt, String authProvider, String accountId, Identity userIdentity, List<Identity> identities, boolean enabled, String userType) {
+        this.jwt = jwt;
+        this.authProvider = authProvider;
+        this.userIdentity = userIdentity;
+        this.accountId = accountId;
+        this.identities = identities;
+        this.enabled = enabled;
+        this.user = userIdentity.getName();
+        this.hostname = null;
+        this.orgs = null;
+        this.teams = null;
+        this.security = null;
+        this.clientId = null;
+        this.userType = userType;
+    }
+
+    public Token(String authProvider, boolean enabled) {
+        this.authProvider = authProvider;
+        this.enabled = enabled;
+        this.jwt = null;
+        this.accountId = null;
+        this.user = null;
+        this.identities = null;
+        this.hostname = null;
+        this.orgs = null;
+        this.teams = null;
+        this.security = null;
+        this.clientId = null;
+        this.userType = null;
+        this.userIdentity = null;
```

Comment:
In all of these constructors, null is the defualt initial value for a field, can you remove all the ones where you are just setting it to null?